appinventor/appengine/src/com/google/appinventor/server/LoginServlet.java

```diff
@@ -203,7 +203,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
       out.println("<h1>" + bundle.getString("setyourpassword") + "</h1>\n");
       out.println("<form method=POST action=\"" + req.getRequestURI() + "\">");
       out.println("<input type=password name=password value=\"\" size=\"35\"><br />\n");
-      out.println("<p></p>");
+      out.println("<p><input type=hidden name=locale value=\""+ locale + "\"></p>");
```

Comment:
`locale` should be escaped here using `StringEscapeUtils.escapeHtml4` otherwise a maliciously constructed locale parameter could be used for XSS attacks.